www-reputation-credentials-protocol.md

```diff
@@ -0,0 +1,265 @@
+the public gossips mechanism.
+
+The credentials are disseminated from the routing hops towards the HTLC senders through onion
+communication channels. The credentials acquisition method and cost are defined by the routing
+hops. For this version of the protocol, we propose upfront fees as a credential acquisition
+method. Payment HTLC are sent to the routing hop and credentials are returned back to the sender
+with a onion reply path. The credentials are counter-signed by the issuer to authenticate their
+origin. The amount of satoshis and the corresponding quantity of credentials delivered is defined
+by each routing hop as a routing policy decision, there is no global pricing of the channel liquidity.
+Alternative acquisiton methods (e.g [proof-of-UTXO ownership](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-November/002884.html))
+can be experimented and deployed in a compatible fashion, without network-wide coordination.
+
+Once a HTLC sender has bootstrapped the possession of sufficient credentials to route across
+the network to reach its payee destination, payment paths can be drawn. For each hop in the path,
+the HTLC onion payload should contain the correct amount of credentials, as requested by the
+hop routing policy.
+
+At reception of a HTLC forward, the credentials signatures should be verified and enforced against
+the routing hop policy. If the HTLC sender has staked enough credentials to lockup outbound liquidity
+for the request msat amount and CTLV duration, the HTLC forward should be committed on the outbound
+channel, and the HTLC setup phase pursued until the destination payee is reached.
+
+At HTLC settlement, if the HTLC is successful, the routing fees are paid accordingly to current
+`channel_update`, a new set of credentials is issued and counter-signed by the hop and returned back
+to the sender through [onion communication channels](https://github.com/lightning/bolts/pull/759).
+A supplement of credentials can be joined, as reward of honoring the routing fees, and unlocking more
+consequential liquidity lockups for future HTLC forwards.
+
+If the HTLC is a failure, the reputation credentials are slashed. No new credentials are issued back
+to the HTLC sender as a punishment. To limit reputation whitewashing, where the jamming damage is superior
+to the reputation acquisition cost, a proportion sould be maintained between the credentials acquisition
+cost (either from upfront fees or honored routing fees) and the channel liquidity lockup rights.
+
+To preserve the confidentiality of the HTLC senders (or payers), the credentials should be anonymized,
+a routing hop should not be able to link between a credential issuance at dissemination and its usage
+at the reception of a HTLC forward request. A cryposystem satisfying those requirements could be to
+leverage EC [blinded signatures](https://sceweb.sce.uhcl.edu/yang/teaching/csci5234WebSecurityFall2011/Chaum-blind-signatures.PDF).
```

Comment:
See the answer on this ML post: https://lists.linuxfoundation.org/pipermail/lightning-dev/2022-November/003756.html

For now, just thinking unlinkability between the blinded credentials and unblinded ones/signatures. In function of if we would like to have secondary transfers of the credentials, or the sender committing to the credentials, we study more advanced cryptosystem. From a UX viewpoint, it might be good to defer the credentials harvesting to the LSPs.

Reply:
Ok. At least good to know that with VOPRF we have one option that seems reasonably well fitting.

Reply:
I see how it can be easily done when credentials are mixed among a set of users (maybe private mixing is hard, but probably not — it's like Tor 2.0 stuff).
But this would be unfortunate, it likely converges to CoinJoin anonymity set stuff, which, again, relies on reputation... Seems like a recursive problem, but maybe I'm missing the fact that we can solve both at the same time somehow?

Now, fancier schemes, I still don't have a good sense of how they should be applied to our problem.
When I think about this, my first question is "what if there was only one client who received the credential, is there a way to protect them?"

1. This is probably solvable either with mixing (see above), or a trusted third party server (like a mix coordinator, but more flexible — and hopefully provably private towards everything) like a watchtower.
2. Maybe this is fine — we assume privacy only if a routing node served 100 clients a day or something. Then, maybe some minimal technology would be sufficient to help those 100 to make sure they get mixed among each other. Possibly something a routing node publishes every day, or maybe even something about burning the collateral of a routing node if cheating happens :P

Reply:
If the credentials do not bind to the sender key material, it should be pretty flexible to have them exchanged in a secondary-market, or even harvested by a LSP. 

I think this is a good point, if the credentials do not bind to the sender key material, or any secret with strict cost to share, they can be exchanged in a cleartext secondary-market. However a cleartext secondary-market would allow the initial credential issuer to collude with the routing hop to deanonymize the "second-hand" HTLC senders. Therefore, if credential transfer is allowed, they should be exchanged in a privacy-anonymizing way mixnet-style. I think this is good to be aware of but a subject for future considerations.